yeat/Snakefile

```diff
@@ -81,22 +82,31 @@ rule downsample:
         read2="seq/fastp/{sample}.R2.fq.gz",
         mash_report="seq/mash/{sample}.report.tsv"
     params:
-        downsample=config["downsample"]
+        coverage=config["coverage"],
+        downsample=config["downsample"],
+        fastp_report="seq/fastp/fastp.json"
     run:
         if params.downsample == -1:
             p = Path("seq/downsample")
             p.mkdir(parents=True, exist_ok=True)
             copyfile(input.read1, output.sub1)
             copyfile(input.read2, output.sub2)
             return
-        result = input.mash_report
-        df = pd.read_csv(result, sep="\t")
+        df = pd.read_csv(input.mash_report, sep="\t")
         genome_size = df["Length"].iloc[0]
-        print(f"genome size: {genome_size}")    
+        print(f"genome size: {genome_size}")
+        data = json.load(open(params.fastp_report))
+        read1_length = data["summary"]["after_filtering"]["read1_mean_length"]
+        read2_length = data["summary"]["after_filtering"]["read2_mean_length"]
+        read_lengths = [read1_length, read2_length]
+        avg_length = sum(read_lengths) // len(read_lengths)
```

Comment:
`avg_length` for all fastq files is compiled here using `fastp`'s report.

Reply:
Here, you're taking a mean of means: the average of R1 average and R2 average. I'd suggest computing the average length directly from the number of bases and reads.

```python
with open(params.fastp_report, "r") as fh:
    qcdata = json.load(fh)
base_count = qcdata["summary"]["after_filtering"]["total_bases"]
read_count = qcdata["summary"]["after_filtering"]["total_reads"]
avg_read_length = base_count / read_count
```